semantics/language/translation/typing/expr.k

```diff
@@ -199,10 +200,19 @@ module C-TYPING-EXPR
      rule typeof(_ ? T:PointerType : _:IntegerType => T)
      rule typeof(_ ? t(Mods:Set, pointerType(T:Type)) : t(Mods':Set, pointerType(T':Type)) => t(Mods Mods', pointerType(compositeType(T, T'))))
           requires areCompatible(stripQualifiers(T), stripQualifiers(T'))
+               andBool notBool (isVariablyModifiedType(T) orBool isVariablyModifiedType(T'))
      rule typeof(_ ? t(Mods1:Set, pointerType(t(Mods1':Set, S:SimpleType))) : t(Mods2:Set, pointerType(t(Mods2':Set, void))) => t(Mods1 Mods2, pointerType(t(Mods1' Mods2', void))))
           requires notBool isFunctionType(t(Mods1', S))
+               andBool notBool isVariablyModifiedType(t(Mods1', S))
      rule typeof(_ ? t(Mods1:Set, pointerType(t(Mods1':Set, void))) : t(Mods2:Set, pointerType(t(Mods2':Set, S:SimpleType))) => t(Mods1 Mods2, pointerType(t(Mods1' Mods2', void))))
           requires notBool isFunctionType(t(Mods2', S))
+               andBool notBool isVariablyModifiedType(t(Mods2', S))
+
+     rule (.K => UNDEF("TTE1",
```

Comment:
Yes, that sounds right, because NULL is a null pointer constant. So the example you constructed so far should actually be correct with the type as you describe. However, if I were to do something like:
```
void foo(int (*x)[*], int y) {
  sizeof(0 ? x : (int (*) [y]) NULL);
}
```

Then it would take the composite type and be undefined.